Add FIPS test case: apache_nssfips.pm

QingmingSu opened the pull request:
Configure Apache server with NSS module, also enable NSSFips

QingmingSu commented:
Successful test run on local openQA:
http://147.2.212.224/tests/392

coolo commented:
IMO it would be worth it to run this test slightly different also without FIPS. I.e. have one apache_nss and one apache_nssfips sharing common code. What do you think?

QingmingSu commented:
Yeah, it makes sense. Which test suite should be added to when the new non-fips case (i.e. apache_nss) ready?

coolo commented:
add it where php5.pm is

QingmingSu commented:
Got it, thanks.
I will create new case apache_nss and make it share most codes with apache_nssfips

coolo commented:
thanks in advance

QingmingSu commented:
Updates:
  - Created lib/apachetest.pm for common codes of apache setup.
  - Added new case apache_nss.pm to test apache with nss module.
  - Added new case apache_nssfips.pm to test apache nss module in FIPS mode.

coolo commented on lib/apachetest.pm:
```diff
@@ -0,0 +1,102 @@
+package apachetest;
+
+use base Exporter;
```
please add a license here too. the rest looks convincing

QingmingSu replied:
Updated to add the missing license statement.

QingmingSu commented:
Oh, I forgot the license statement for the library file.
The copyright has been added now.

The test cases have been tested in my local openQA:
--For non-fips cases: apache_ssl and apache_nss
http://147.2.212.224/tests/410

--For fips cases: apache_ssl and apache_nssfips
http://147.2.212.224/tests/411
